### PR TITLE
comment --pr=true by default

### LINKED ISSUE #1139: comment --pr=true default
https://github.blog/changelog/2022-08-04-commit-comments-no-longer-appear-in-the-pull-request-timeline/ meaning the `--pr` flag is now mandatory for CML reports to show up in PRs. Kinda a UX breaking change potentially.

Options (pick just one):

1. update docs & examples, adding `--pr` everywhere (breaking change, https://github.com/iterative/cml.dev/issues/331)

<details><summary>also auto-detect this/print warnings?</summary>

```bash
$ cml comment ...
(ok)
$ cml pr ...
(ok)
$ cml comment ...
WARNING: did you forget `--pr` flag?
```

</details>

2. `cml comment ... [--pr=true]` by default (UX friendly)
   + iff not in a PR, print a warning and fallback to `--pr=false` (commit-comment)
   + users need explicit `--pr=false`/`--no-pr` to use commit-comments in PRs (edge case)
3. `cml comment ... [--pr=auto]` by default (better fail-not-warn error handling)
   + try to PR-comment, fallback to commit-comment
   + explicit `--pr={true,false}` throws error on failure (no fallback)
   + however, is this implementation possible? does yargs allow `--flag=foo` == `"foo"` && `--flag` == `true` (latter needed for backwards compat)?

### PR DESCRIPTION
**Pull request by tasdomas**
Closes #1139.

### CODE REVIEW

**Comment by tasdomas** [comment]
The more I look at this, the less it's making sense - marking as draft for now.

**Review by casperdcl — CHANGES_REQUESTED** [review]
was thinking something more like this.

but my tests show it's not quite fully backwards-compatible:

- works: `cml comment create report.md --pr`
- works: `cml comment create --pr --publish report.md`
- works: `cml comment create report.md --pr=auto`
- fails: `cml comment create --pr report.md` (interprets as `--pr=report.md` and complains about no positional arg)

File: bin/cml/comment/create.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ exports.options = kebabcaseKeys({
   pr: {
     type: 'boolean',
     description:
-      'Post to an existing PR/MR associated with the specified commit'
+      'Post to an existing PR/MR associated with the specified commit',
+    default: true
```
[the remark this answers is not in the document]
Comment (tasdomas) [reply bin/cml/comment/create.js:18]: Should pr number also be a valid choice?

**Comment by tasdomas** [comment]
Closing, as this is not the solution we're looking for.

**Pull request closed without merging**